common/interpolation/include/interpolation/spline_interpolation.hpp

```diff
@@ -40,9 +45,86 @@ struct MultiSplineCoef
   std::vector<double> d;
 };
 
+// static spline interpolation functions
 std::vector<double> slerp(
   const std::vector<double> & base_keys, const std::vector<double> & base_values,
   const std::vector<double> & query_keys);
+
+// std::vector<double> slerpDiff(
+//   const std::vector<double> & base_keys, const std::vector<double> & base_values,
+//   const std::vector<double> & query_keys);
+
+// TODO(murooka) use template
+// template <typename T>
+// std::vector<double> slerpYawFromPoints(const std::vector<T> & points);
+std::vector<double> slerpYawFromPoints(const std::vector<geometry_msgs::msg::Point> & points);
```

Comment:
![image](https://user-images.githubusercontent.com/20228327/153788343-c3b24013-3cc6-457e-930e-81ff3c0523d3.png)

Reply:
spline interpolation classes (SplineInterpolation1d, SplineInterpolationPoint2d) use some common functions like getSplineCoefficients, getSplineInterpolatedValues.
I think we do not have to write a test for these common functions (and of course I do not want to write more tests), so I put them in anonymous namespace, and put all spline interpolation classes in the same file.

What do you think? Should I add a test for common functions to put SplineInterpolationPoint2d in another file?

Reply:
As Horibe-san pointed, SplineInterpolationPoint2d may not be propper. Path2d seems like interpolation for path points (not trajectory points)
This class does not depend on any types like Path, so how about SplineInterpolation"Points2d"?

Reply:
@TakaHoribe

Reply:
I could seperated them, and created spline_interpolation_points_2d.hpp. How about this change?
https://github.com/autowarefoundation/autoware.universe/blob/a683cc0b430221aad789e06c5843397a65bad5f6/common/interpolation/include/interpolation/spline_interpolation_points_2d.hpp

Reply:
LGTM!